File: examples/documents/stringset-ops.ts
import type { Task } from "../_harness/generated/ts/Task.generated";

// Stringset fields hold a set of strings (tags, labels): mutate with
// add/remove, test membership, convert to an array for display.
export function stringsetOps(task: Task) {
  // #region example
  // Add/remove tags
  task.tags?.add("urgent");
  task.tags?.remove("low-priority");

  // Check membership
  if (task.tags?.has("urgent")) {
    // ...
  }

  // Convert to array for display
  const tagList = task.tags?.toArray() ?? [];
  // #endregion example
  return tagList;
}

File: examples/documents/stringset-ops.swift
import JsBaoClient

// Stringset fields are a native Set<String>: mutate the set, test
// membership, then save the record to persist the change.
func stringsetOps(documentId: String, taskId: String) async throws {
  // #region example
  if var task = try await Task.find(taskId) {
    // Add/remove tags
    task.tags?.insert("urgent")
    task.tags?.remove("low-priority")

    // Check membership
    if task.tags?.contains("urgent") == true {
      // ...
    }

    try task.save(in: documentId)
  }
  // #endregion example
}
